domain/src/main/java/edu/nextstep/camp/calculator/domain/Calculator.kt

```diff
@@ -0,0 +1,56 @@
+package edu.nextstep.camp.calculator.domain
+
+class Calculator {
+
+    fun evaluate(expression: String?): Int {
+        if (expression.isNullOrBlank())
+            throw IllegalArgumentException("null 또는 빈 공백 문자열은 수식이 아닙니다.")
+
+        return evaluateSymbols(splitToSymbols(expression))
+    }
+
+    private fun splitToSymbols(expression: String): List<Symbol> {
+        return expression.split(' ')
+            .map { symbol ->
+                when (symbol) {
+                    "+" -> Symbol.Sign.PLUS
+                    "-" -> Symbol.Sign.MINUS
+                    "*" -> Symbol.Sign.TIMES
+                    "/" -> Symbol.Sign.DIVISION
+                    else -> symbol.toIntOrNull()?.let {
+                        Symbol.Number(it)
+                    } ?: throw IllegalArgumentException("수식에는 +, -, *, /만 사용될 수 있습니다.")
+                }
+            }
+    }
+
+    private fun evaluateSymbols(symbols: List<Symbol>): Int {
+        var sign: Symbol.Sign? = null
+
+        return symbols.fold(Symbol.Number(0)) { prevResult, symbol ->
+            when (symbol) {
+                is Symbol.Sign -> {
+                    sign = symbol
+                    prevResult
+                }
+                is Symbol.Number -> {
+                    evaluateOne(prevResult, sign, symbol)
+                }
+            }
+        }.n
+    }
```

Comment:
컬렉션 API와 고차 함수를 활용하는 부분은 좋았습니다만 외부의 변수(sign)에 어떠한 값을 저장시키는 행위는 사이드 이펙트를 유발시킬 수 있기에 이점을 누릴수 있는 구조는 아니라고 생각합니다. 차라리 일반 for 문으로 작성하는 편이 더 낫다고 생각됩니다.

```suggestion

    private fun evaluateSymbols(symbols: List<Symbol>): Int {
        val firstNumber = symbols.first() as Symbol.Number
        val remainSymbols = symbols.drop(1)
        
        return remainSymbols.chunked(2)
            .fold(firstNumber) { prevResult, (sign, symbol) ->
                evaluateOne(prevResult, sign as Symbol.Sign, symbol as Symbol.Number)
            }.n
    }
```

수정을 요청드리는건 아니고 현재 구조에서 컬렉션 API를 사용해 사이드 이펙트 없이 구현한다면 이렇게 했을것 같다 정도로 참고해주시면 좋을 것 같습니다. 🤔